microprofile.ls/org.eclipse.lsp4mp.ls/src/main/java/org/eclipse/lsp4mp/ls/java/JavaFileTextDocumentService.java

```diff
@@ -125,8 +125,17 @@ public void didClose(DidCloseTextDocumentParams params) {
 
 	@Override
 	public void didSave(DidSaveTextDocumentParams params) {
+		// Do nothing..
+		
+		// Since the beginning of the MP LS project,
+		// saving a Java file triggers validation for all open Java files,
+		// which can negatively impact performance.
+		// As there's no comment explaining the reason for this behavior,
+		// we are disabling this feature.
+
 		// validate all opened java files which belong to a MicroProfile project
-		triggerValidationForAll(null);
+		// triggerValidationForAll(null);

```

Comment:
I think this breaks the following behaviour. You have a Java source file that defines configuration properties used in the configuration file. If those are added/removed, it affects some diagnostics :

https://github.com/user-attachments/assets/2a21b852-6686-465b-8a79-9f2addccff27

The diagnostics will not update when saving the config file, but they do update when continuing to type in the source file, or re-opening it. Are we ok with that ?

Are we ok with that ?

Reply:
I remember that I had the similar problem with IJ Quarkus.

I am pretty sure that if you save your application.properties a second time, it will work.

If it that it is because we load in cache properties (for checking validation on Java file) from the /target/application.properties and not from src/main/resources/application.properties. 

I think we should load cache from src/main/resources/application.properties. 

We will loose the computed variable from maven for instance declared in teh properties file, but I have never seen this usecase.

Reply:
Saving a second time will not fix this. Validation isn't triggered. You either need to type, or re-open the file.

Reply:
That's strange it is working for me?

Reply:
Are you making changes and then saving or just hitting "Save" on an unchanged file ? Making changes will update the diagnostics but saving doesn't. Where would the validation come from when `triggerValidationForAll` is removed from `didSave` above.

Reply:
Typing a space in the stale Java source file even without the save will update the diagnostics. Look at the video "With this change" above! You can see just typing updates the diagnostics so that's fine.

Reply:
Sorry I mean tyep a space and save application.properried and do that twice.

It should update java diagnostics correctly, right?

Reply:
You're right, it does work. Interesting.